controller/src/main/java/io/pravega/controller/store/stream/InMemoryStream.java

```diff
@@ -93,46 +89,62 @@ public void refresh() {
     }
 
     @Override
-    CompletableFuture<Void> checkStreamExists(Create create) {
-        CompletableFuture<Void> result = new CompletableFuture<>();
+    CompletableFuture<CreateStreamResponse> checkStreamExists(StreamConfiguration configuration, long timestamp) {
+        CompletableFuture<CreateStreamResponse> result = new CompletableFuture<>();
+
+        final long time;
+        final StreamConfiguration config;
+        final Data<Integer> currentState;
+        synchronized (lock) {
+            time = creationTime.get();
+            config = this.configuration;
+            currentState = this.state;
+        }
+
+        if (time != Long.MIN_VALUE) {
```

Comment:
I think this would be a lot easier to read if the block below were moved into a private helper method which returned a CreateStreamResponse. That would allow all of these special cases to be handled by if(...) {return ...} so as not to require nested ifs